xsuite/src/world/csworld.test.ts

```diff
@@ -0,0 +1,801 @@
+import { beforeEach, expect, test, beforeAll, afterAll, assert } from "vitest";
+import { assertAccount, assertVs } from "../assert";
+import { e } from "../data";
+import { Tx } from "../proxy";
+import { DummySigner } from "./signer";
+import {
+  generateContractU8AAddress,
+  generateWalletU8AAddress,
+  isContractAddress,
+} from "./utils";
+import { CSWorld, CSContract, CSWallet } from ".";
+
+let world: CSWorld;
+let wallet: CSWallet;
+let otherWallet: CSWallet;
+let contract: CSContract;
+const fftId = "FFT-abcdef";
+const sftId = "SFT-abcdef";
+const worldCode = "file:contracts/world/output/world.wasm";
+const zeroBechAddress =
+  "erd1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq6gq4hu";
+const zeroHexAddress =
+  "0000000000000000000000000000000000000000000000000000000000000000";
+const zeroBytesAddress = new Uint8Array(32);
+const emptyAccount = {
+  nonce: 0,
+  balance: 0,
+  code: "",
+  codeHash: "",
+  codeMetadata: "",
+  owner: "",
+  kvs: {},
+};
+const explorerUrl = "http://explorer.local";
+
+beforeAll(async () => {
+  world = await CSWorld.start({
+    explorerUrl,
+    // verbose: true,
+    // debug: true,
+    waitFor: 120_000,
+  });
+  wallet = await world.createWallet({
+    balance: 10n ** 18n,
+    kvs: [e.kvs.Esdts([{ id: fftId, amount: 10n ** 18n }])],
+  }); // wallet in shard 0
+  otherWallet = await world.createWallet();
+
+  // pass an epoch so system smart contracts are enabled
+  await world.generateBlocksUntilEpochReached(1);
```

Comment:
If I remember correctly, the chain simulator can't be started at epoch 1 at the moment for the moment, right? If yes, then I will push to allow the chain simulator to start at epoch 1, or to make the system smart contract enabled at epoch 0 (the latter is what makes more sense).

Reply:
The comment was incorrect, the reason we need epoch 1 is because that's the epoch at which Smart Contract deploys are enabled.
Although now with the new `nodeOverride.toml` in the Chain Simulator I can probably overwrite that so it works from epoch 0, will check.

Reply:
Done, `nodeOverride.toml` file was used instead